LaTeX/LaTeX.sublime-syntax

```diff
@@ -198,6 +207,123 @@ contexts:
     - include: paragraph-pop
     - include: else-pop
 
+  xparse-argspec:
+    - match: (?=\{)
+      set:
+        # deliberately using an anonymous scope here. we need two match the `{`
+        # in two stages, so we can apply the `clear_scopes`, and still be able
+        # to push `xparse-argspec` onto the stack without messing up scoping.
+        - clear_scopes: 1
+        - match: '\{'
+          scope: punctuation.definition.group.brace.begin.latex
+          set: xparse-argspec-content
+
+    - include: paragraph-pop
+    - include: else-pop
+
+  xparse-argspec-content:
+    - clear_scopes: 1
+    - meta_scope: meta.function.parameters.latex
+    - include: brace-group-end
+    - include: macro-braces
+    - include: xparse-single-argspec
+    # not actually allowed, but included here to future-proof highlighting
+    - include: general-constants
+    - include: general-commands
+    - include: paragraph-pop
+
+  xparse-single-argspec:
+    # modifiers and simple specs
+    - match: '[+!movsb]'
+      scope: string.unquoted.latex meta.annotation.parameters.latex
+
+    - match: t
+      scope: string.unquoted.latex meta.annotation.parameters.latex
```

Comment:
I guess those are flags, just like shellscript flags (e.g.: `cmd -f`)?

In that case we could/should probably use the same scope as Bash...

```suggestion
      scope: meta.parameter.option.latex variable.parameter.option.latex
```

Reply:
I'm not sure if that is an adequate comparison. (Almost) each letter represents one of the arguments that can be passed to the function that is being defined, e.g. `m` indicates that there is a mandatory argument, `o` indicates that there is an optional argument that needs to be wrapped in `[]`, `t` indicates a boolean argument that checks for the occurrence of a specific token. 
The exceptions are `+!`, which are modifiers that apply to the next letter that specifies an argument.

Reply:
So those are more like data types or better modifiers (`storage.type`, `storage.modifier`)?

Reply:
Those are actually types (see: https://mirror.informatik.hs-fulda.de/tex-archive/macros/latex/contrib/l3packages/xparse.pdf page 2).

Reply:
well, that really depends on how you understand the word "type". In python, would you consider the two arguments with name `x` to be of different type? 
```
f(x)
g(*, x)
```
it is a specifier of how arguments to this function are picked up.

Reply:
There's no perfect matching scope for it as this kind of argument list specification seems somewhat unique to LaTeX. We could invent one or interpret existing ones in ways which match the language scheme. Maybe using `storage.modifier` for all of them would be fine, too, or `keyword.declaration` or `keyword.other` if in doubt.

I am just not convinced of `string` scope. `meta.annotation` is reserved for decorators and attributes (e.g.: C#).